content/fr/articles/js/es2015/arrow-functions.md

```diff
@@ -0,0 +1,168 @@
+---
+date: "2015-12-15"
+title: "ES6, ES2015 : Les fonctions fléchées"
+tags:
+  - JavaScript
+  - ES6
+  - ES2015
+authors:
+  - MoOx
+---
+
+ES2015 nous apporte encore du sucre syntaxique qui risque très probablement de
+vous faire oublier ``Function.prototype.bind()``.
+
+Les fonctions fléchées offrent une syntaxe raccourcie des fonctions en utilisant
+la syntaxe `=>`.
+
+```js
+// es5
+var myFn = function(x) {
+    return x + 1
+}
+
+// es6
+const myFn = (x) => {
+  return x + 1
+}
+```
+
+Les fonctions fléchées sont syntaxiquement similaire à ce qu'on trouve déjà dans
+d'autres langages comme CoffeeScript, Java (8+), C#…
+
+Elles supportent les expressions ou les en tant que corp de fonction.
+Dans notre exemple précédent, nous avons utilisé un corps classique.
+Pour de simple fonctions, nous pouvons utiliser des expressions, le but étant
+d'avoir quelque chose de concis.
+Notre exemple précédent peut ainsi s'écrire de la sorte :
+
+```js
+const myFn = (x) => x + 1
+```
+
+Notez que lorsque l'on a qu'un argument dans notre fonction, on peut omettre les
+parenthèses. Notre exemple peut donc aussi s'écrire ainsi :
+
+```js
+const myFn = x => x + 1
+```
+
+Si notre expression mérite d'être sur plusieurs lignes, on peut aussi utiliser des
+parenthèses :
+
+```js
+const myFn = x => (
+  x +
+  1 // Ici vous pouvez faire du multi lignes tranquille, on peut imaginer du JSX
+)
+```
+
+Les exempls suivants sont donc tous les mêmes :
+
+```js
+const myFn = (x) => {
+  return x + 1
+}
+// ===
+const myFn = (x) => x + 1
+// ===
+const myFn = x => x + 1
+// ===
+const myFn = x => (x + 1)
+```
+
+Dans la pratique vous utiliserez souvent cette syntaxe pour des petites méthodes
+comme par exemple lors de l'utilisation des fonctions de tableau
+reduce/filter/map etc.
+
+```js
+const nums = [1, 2, 3, 4, 5]
+const odds = nums.filter(v => v%2) // [1, 3, 5]
+const oddsSum = odds.reduce((sum, v) => sum+v, 0) // 9
+```
+
+## Les fonctions fléchées n'ont pas de `this`
+
+Oui, vous avez bien lu : à l'inverse des fonctions normales, les fonctions
+fléchées partagent le même ``this`` lexical que leur code autour.
```

Comment:
`code autour`c'est une tournure un peu étrange. 


Oui, vous avez bien lu : à l'inverse des fonctions normales, les fonctions
fléchées ne redéfinissent pas le `this `?